overlord/devicestate/firstboot.go

```diff
@@ -415,10 +418,30 @@ var loadDeviceSeed = func(st *state.State, sysLabel string) (deviceSeed seed.See
 		return assertstate.AddBatch(st, batch, nil)
 	}
 
-	if err := deviceSeed.LoadAssertions(assertstate.DB(st), commitTo); err != nil {
+	db := assertstate.DB(st)
+	if err := deviceSeed.LoadAssertions(db, commitTo); err != nil {
 		return nil, err
 	}
 
+	// if model is dangerous, check if there is auto-import.assert and import it
+	if deviceSeed.Model().Grade() == asserts.ModelDangerous {
+		seed20AssertionsLoader, ok := deviceSeed.(seed.AutoImportAssertionsLoaderSeed)
+		if !ok {
+			logger.Noticef("failed to cast seed20 assertion loader")
+			return deviceSeed, nil
+		}
+		err = seed20AssertionsLoader.LoadAutoImportAssertions(commitTo)
+		if err != nil {
+			logger.Noticef("failed to load auto import assertions: %v", err)
+			return deviceSeed, nil
+		}
+		serial, _ := findSerial(st, nil)
+		_, userError := createAllSystemUsers(st, db, deviceSeed.Model(), serial, true)
```

Comment:
Is it necessary for `userError` to be a new variable here? can't we reuse `err` ?

Reply:
it was from before function was returning custom error struct, indeed now it can reuse err. Fixed